### ISSUE TITLE
BZ2Source and ZipSource not usable on Python 2.6

### ISSUE THREAD

**Issue by klada** [alimanfoo/petl#315]
When reading CSV data, the source is being used as a context manager (opened with `with` statement). 
However the implementations of BZ2File and ZipFile in Python 2.6 do not support this, which raises an an AttributeError:

```
  File "/usr/lib/python2.6/site-packages/petl/io/csv_py2.py", line 42, in __iter__
    with self.source.open('rb') as buf:
AttributeError: 'bz2.BZ2File' object has no attribute '__exit__'
```

**Comment by alimanfoo** [comment]
Many thanks for the bug report, I'll look into a workaround.

On Friday, 20 February 2015, Dan Klaffenbach <notifications@github.com>
wrote:

> When reading CSV data, the source is being used as a context manager
> (opened with with statement).
> However the implementations of BZ2File and ZipFile in Python 2.6 do not
> support this, which raises an an AttributeError:
>
>   File "/usr/lib/python2.6/site-packages/petl/io/csv_py2.py", line 42, in __iter__
>     with self.source.open('rb') as buf:
> AttributeError: 'bz2.BZ2File' object has no attribute '__exit__'
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/alimanfoo/petl/issues/315>.
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Web: http://purl.org/net/aliman
Email: alimanfoo@gmail.com
Tel: +44 (0)1865 287721

**Comment by alimanfoo** [comment]
Fix included in v1.0.6.
